Assets/Scripts/Audio/AudioSpectrumDrawer.cs

```diff
@@ -0,0 +1,90 @@
+using ProefExamen.Audio.WaveFormDrawer;
+using UnityEngine;
+
+namespace ProefExamen.Audio.SpectrumDrawer
+{
+    public class AudioSpectrumDrawer : MonoBehaviour
+    {
+        [SerializeField]
+        private int _audioSamples = 512;
+
+        [SerializeField]
+        private float _lerpAmount = 20;
+
+        [SerializeField]
+        private float _maxScale = 1200;
+
+        [SerializeField]
+        private Vector3 _offsetPosition = Vector3.zero;
+
+        [Space]
+        [SerializeField]
+        private GameObject _imagePrefab;
+
+        private GameObject[] _visualizers;
+        private AudioWaveformDrawer waveformDrawer;
+
+        private float[] _samples;
+
+        private void Awake()
+        {
+            _samples = new float[_audioSamples];
+            waveformDrawer = FindObjectOfType<AudioWaveformDrawer>();
+        }
+
+        /// <summary>
+        /// Method which analyses the specified audio clip's samples.
+        /// </summary>
+        /// <param name="clip">Clip used for analysing the audio spectrum data.</param>
+        public void VisualizeSongSpectrum(AudioClip clip)
+        {
+            _visualizers = null;
+            foreach (Transform child in transform)
+                Destroy(child.gameObject);
+
+            waveformDrawer.InitializeDrawer(clip);
+            _visualizers = new GameObject[_samples.Length];
+
+            CreateBandVisualizers();
+        }
+
+        /// <summary>
+        /// Method used for creating band visualizers.
+        /// </summary>
+        private void CreateBandVisualizers()
+        {
+            for (int i = 0; i < _visualizers.Length; i++)
+            {
+                GameObject objToSpawn = Instantiate(_imagePrefab, transform);
```

Comment:
```suggestion
                GameObject newBandVisualizer = Instantiate(_imagePrefab, transform);
```
Something like that; objToSpawn isn't accurate of what the variable contains since it's already spawned.